### PR TITLE
zypper ref before reinstalling traditional stack packages

### PR DESCRIPTION
**Pull request by Bischoff**
## What does this PR change?

We have such errors
```
Feature:Migrate a traditional client into a Salt minion
Scenario:Cleanup: register minion again as traditional client
And I install the traditional stack utils on "sle_client"

FAIL: zypper --non-interactive install -y spacewalk-client-tools spacewalk-check spacewalk-client-setup 
  mgr-daemon mgr-osad mgr-cfg-actions returned 8. output : Loading repository data...
(...)
File './noarch/python3-spacewalk-client-tools-4.1.8-411.2.13.devel41.noarch.rpm' not found on medium
```
that can happen when the tools are rebuilt while we test.

This PR does a `zypper refresh` (or equivalent) just before reinstalling traditional stack packages.

## Links

Ports:
* 4.0: SUSE/spacewalk#13791 SUSE/spacewalk#13803
* 4.1: SUSE/spacewalk#13790 SUSE/spacewalk#13802

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

### CODE REVIEW

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:799]: Please, keep it in a unique step definition.

**Review by srbarrios — CHANGES_REQUESTED** [review]
Please keep it in a unique step definition as it is now.

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [inline testsuite/features/step_definitions/command_steps.rb:799]:
```
When(/^I (install|remove) the traditional stack utils (on|from) "([^"]*)"$/) do |action, where, host|
  pkgs = 'spacewalk-client-tools spacewalk-check spacewalk-client-setup mgr-daemon mgr-osad mgr-cfg-actions'
  step %(I refresh the metadata for "#{host}") if action == 'install'
  step %(I #{action} packages "#{pkgs}" #{where} this "#{host}")
end
```

Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:799]: The code for install and remove is different.

It is ridiculous to have it in an unique step just to dijoin cases in that unique step.

Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:799]: We do not refresh metadata before removing packages.

Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:799]: I'm sorry but I don't agree. 
Please if we need a refresh of the metadata only when installing, express it as I shared bellow.
```
  step %(I refresh the metadata for "#{host}") if action == 'install'
```

Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:799]: Not elegant.

Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:799]: I don't find elegant code duplication. And I consider it a bad practice.
If you don't probe a good reason to make this refactor mandatory, I would like that we only add the oneliner I suggested.

Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:799]: I will do it, but only to have peace...

I don't understand this compulsion for intricated and complex code.

Comment (Bischoff) [reply testsuite/features/step_definitions/command_steps.rb:799]: I gave a whole series of good reasons in previous comment.

**Review by srbarrios — APPROVED** [review]
Approving, so I don't block this PR if I'm not around :)

File: testsuite/features/step_definitions/command_steps.rb
Comment (srbarrios) [reply testsuite/features/step_definitions/command_steps.rb:799]: Thank you

**Comment by Bischoff** [comment]
> Approving, so I don't block this PR if I'm not around :)

Thank you.

**Review by calancha — APPROVED** [review]
Good catch! This makes our test suite more robust.

**Review by lkotek — APPROVED** [review]
LGTM.

**Comment by juliogonzalez** [comment]
Since this affected head and it's simple, let's merge.

**Pull request merged**
